js/components/common/react_modal.js.jsx

```diff
@@ -92,17 +178,43 @@ class Description extends React.Component {
         <p>
           <strong>Timetable: </strong>
         </p>
-        {this.props.sessions.map(function (lecture, i) {
-          return (
-            <p key={i}>
-              {lecture.meetData.code +
-                lecture.meetData.session +
-                "-" +
-                lecture.meetData.section}
-            </p>
+
+        {Object.keys(this.props.sessions).map(session =>
+          this.props.sessions[session].length !== 0 ? (
+            <div key={session}>
+              <h2>{this.props.course.name + "-" + session}</h2>
+              <div className="ag-theme-alpine" style={{ height: 500, width: 940 }}>
+                <AgGridReact
+                  rowData={this.props.sessions[session]}
+                  columnDefs={[
+                    { field: "Activity", width: 130 },
+                    { field: "Instructor", width: 170 },
+                    { field: "Availability" },
+                    { field: "WaitList", width: 120 },
+                    {
+                      field: "Time",
+                      cellStyle: { whiteSpace: "pre" },
+                      cellRenderer: param => this.cellRendererBreak(param.data.Time),
```

Comment:
You don't need to define a helper method here. Use the built-in `Array.prototype.join`.